### PR TITLE
Update CSI Spec dependency to v1.0.0-rc2

### PR DESCRIPTION
**Pull request by davidz627**
/assign @msau42 @saad-ali 

xref https://github.com/kubernetes/kubernetes/pull/71020

### CODE REVIEW

File: pkg/common/utils.go [hunk 1]
```diff
@@ -133,3 +136,11 @@ func GetDeviceName(volKey *meta.Key) (string, error) {
 func CreateNodeID(project, zone, name string) string {
 	return fmt.Sprintf(nodeIDFmt, project, zone, name)
 }
+
+func TimeToTimestamp(t time.Time) *timestamp.Timestamp {
```
Comment (saad-ali) [inline pkg/common/utils.go]: Use the `github.com/golang/protobuf/ptypes` library. It has a method to do this correctly: `func TimestampProto(t time.Time) (*tspb.Timestamp, error)`

**Comment by davidz627** [comment]
/hold

Needs to be updated again after csi-test is cut

File: pkg/gce-pd-csi-driver/controller.go [hunk 2]
```diff
@@ -152,7 +152,7 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 	content := req.GetVolumeContentSource()
 	if content != nil {
 		if content.GetSnapshot() != nil {
-			snapshotId = content.GetSnapshot().GetId()
+			snapshotId = content.GetSnapshot().GetSnapshotId()
```
Comment (saad-ali) [inline pkg/gce-pd-csi-driver/controller.go]: Add a TODO or open a bug to add support for "Volume Source" (cloning) which is new in 1.0.

File: pkg/gce-pd-csi-driver/controller.go [hunk 3]
```diff
@@ -152,7 +152,7 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
...
 		}
 	}
```
Comment (saad-ali) [inline pkg/gce-pd-csi-driver/controller.go]: Modify `ControllerServiceCapability` to return new `PUBLISH_READONLY` capability.

See https://github.com/container-storage-interface/spec/pull/310/files for details

**Comment by davidz627** [comment]
@saad-ali resolved your comments

**Comment by davidz627** [comment]
I don't think the snapshot implementation is broken, I've done a small refactor to ensure that it works with the new spec. @jingxu97 @xing-yang can verify at some later point

**Comment by saad-ali** [comment]
@davidz627 Got it, thanks for the clarification.

**Comment by davidz627** [comment]
Final TODOs before unholding:

- [ ] Update all sidecar containers to v1.0.0 (in dev or base?)
- [ ] Verify everything works end to end on k8s

**Pull request merged**